File: Archivos Visual Studios/entornos desarrollo/C++/ejercicio2/EstructurasSeleccion/ejercicio3.cpp
/*
* Nombre: switch-case
*Autor: Joshua Sangareau
*Fecha: 08/03/2024
*Descripción: Realiza un programa donde usuario ingresa un número del mes y luego
se utiliza la declaración switch para seleccionar el nombre del mes
correspondiente. Cada case representa un número de mes y se imprime
el nombre del mes correspondiente. Si el número ingresado no está en el
rango de 1 a 12, se imprime un mensaje de error utilizando default
*/

#include <iostream>

using namespace std;

int main(){

    int numeroMes;

    cout << "Introduzca un numero de mes" << endl;
    cin >> numeroMes;

    switch (numeroMes){
        case (1):{
            cout << "El mes es enero" << endl;
            break;
        }case (2):{
            cout << "El mes es febrero" << endl;
            break;
        }case (3):{
            cout << "El mes es marzo" << endl;
            break;
        }case (4):{
            cout << "El mes es abril" << endl;
            break;
        }case (5):{
            cout << "El mes es mayo" << endl;
            break;
        }case (6):{
            cout << "El mes es junio" << endl;
            break;
        }case (7):{
            cout << "El mes es julio" << endl;
            break;
        }case (8):{
            cout << "El mes es agosto" << endl;
            break;
        }case (9):{
            cout << "El mes es septiembre" << endl;
            break;
        }case (10):{
            cout << "El mes es octubre" << endl;
            break;
        }case (11):{
            cout << "El mes es noviembre" << endl;
            break;
        }case (12):{
            cout << "El mes es diciembre" << endl;
            break;
        }default:{
            cout << "Error: el numero de mes es incorrecto, debe ser del 1 al 12" << endl;
            break;
        }
    }
    return 0;
}

File: Programación/Documentos Eclipse-Java/SangareauJoshua_4.15_metodos/src/ejercicio3/Ejercicio3.java
/*
 * Crea un método llamado indicaMesDias que devolverá una cadena de texto.
 El método recibirá un valor entero.
 Comprobará que el valor oscile entre 1 y 12 (ambos inclusive)
En caso contrario devolverá un mensaje indicando que el mes es incorrecto.
 En caso de ser correcto devolverá un mensaje indicando el mes al que se refiere y el
número de días del mes: Ejemplo: " Enero, 31 días".

 */
package ejercicio3;

import java.util.Scanner;

public class Ejercicio3 {

	public static void main(String[] args) {
		// ----------------------------------------------
		// Declaración de variables
		// ----------------------------------------------

		// Constantes
		// Variables de entrada
		int mes;
		// Variables de salida
		// Variables auxiliares
		// Clase Scanner para petición de datos

		Scanner teclado = new Scanner(System.in);

		// ----------------------------------------------
		// Entrada de datos
		// ----------------------------------------------

		System.out.println("Introduzca el número del mes: ");
		mes = teclado.nextInt();

		// ----------------------------------------------
		// Procesamiento
		// ----------------------------------------------

		System.out.println(indicaMesDias(mes));

	}

	public static String indicaMesDias(int x) {
		Scanner teclado = new Scanner(System.in);
		String mes = null;

		switch (x) {
		case 1: {
			mes = "Enero, 31 días";
			break;
		}
		case 2: {
			mes = "Febrero, 28 días";
			break;
		}
		case 3: {
			mes = "Marzo, 31 días";
			break;
		}
		case 4: {
			mes = "Abril, 30 días";
			break;
		}
		case 5: {
			mes = "Mayo, 31 días";
			break;
		}
		case 6: {
			mes = "Junio, 30 días";
			break;
		}
		case 7: {
			mes = "Julio, 31 días";
			break;
		}
		case 8: {
			mes = "Agosto, 31 días";
			break;
		}
		case 9: {
			mes = "Septiembre, 30 días";
			break;
		}
		case 10: {
			mes = "Octubre, 31 días";
			break;
		}
		case 11: {
			mes = "Noviembre, 30 días";
			break;
		}
		case 12: {
			mes = "Diciembre, 31 días";
			break;
		}
		default: {
			mes = "El número del mes introducido es incorrecto, vuelva a introducirlo ";
		}

		}

		return mes;

	}
}